fem/coefficient.hpp

```diff
@@ -1426,12 +1426,13 @@ class SumCoefficient : public Coefficient
 class SymmetricMatrixCoefficient : public MatrixCoefficient
 {
 protected:
+
    /// Internal matrix used when evaluating this coefficient as a DenseMatrix.
-   DenseSymmetricMatrix mat;
+   mutable DenseSymmetricMatrix mat_aux;
```

Comment:
minor: maybe `mat_sym`  as it is a bit more descriptive ?  (just a personal preference so feel free to ignore)

Reply:
It is `mat` in child classes vs. `mat_aux` here, where both are symmetric, so it would be rather confusing renaming this one to `mat_sym`. I would keep this, because it indicates it is just a mutable auxiliary thing, which no one should rely on as it changes all the time.